TypeScriptActivity/typeScriptBasics.ts

```diff
@@ -0,0 +1,121 @@
+// 1.Basic Types
+let booleanVar: boolean = true;
+let numVar: number = 123;
+let stringVar: string = "Hello World!";
+let stringArrayVar: string[] = ["One", "Two", "Three", "Four", "Five"];
+let tuple: [string, number, boolean] = ["Name", 1, true];
+
+// 2.Functions
+function addTwoNumbers(num1: number, num2: number): number {
+    return num1 + num2;
+}
+console.log(addTwoNumbers(10, 2));
+
+// 3.Interfaces
+interface Person {
+    firstName: string,
+    lastName: string
+}
+
+function generateFullName(obj: Person): string {
+    return obj.firstName + " " + obj.lastName;
+}
+
+let user: Person = {
+    firstName: "John",
+    lastName: "Doe"
+}
+console.log(generateFullName(user));
+
+// 4. Classes
+class Student implements Person {
+
+    public constructor(public firstName: string, public lastName: string) { }
+
+    public generateFullNameofUser(): string {
+        return this.firstName + " " + this.lastName;
+    }
+
+}
+
+let student = new Student("John", "Doe");
+console.log(student.generateFullNameofUser());
+
+// 6. Union and Intersection Types
+function unionExample(parameter: number | string) {
+    if (typeof parameter == "number")
+        return parameter;
+    return parameter.length;
+}
+
+console.log(30);
+console.log("Hello World!");
+
+// 5.Generics
+function identity<S>(value: S): S {
+    return value;
+}
+
+console.log(identity<boolean>(true));
+console.log(identity<string>("true"));
+
+class GenericNumber<T> {
+
+    private sum!: number;
+
+    constructor(private num1: number, private num2: number) { }
+
+    initialiseSumToZero() {
+        this.sum = 0;
+    }
+
+    addTwoNumbers() {
+        this.initialiseSumToZero();
+        this.sum =  this.num1 + this.num2;
+        return this.sum;
+    }
+}
+
+// 7.Decorators
+@Logger
```

Comment:
What is the purpose of decorator

Reply:
Decorators are used to overwrite the behaviour of a function or add additional functionalities to the class/function.

Spread Operator (...) can be used to create a shallow copy of an object and not a deep copy of the object.